kernel/src/table_properties/deserialize.rs

```diff
@@ -0,0 +1,287 @@
+use crate::expressions::ColumnName;
+
+use std::collections::HashMap;
+use std::time::Duration;
+
+use serde::de::{self, DeserializeSeed, Deserializer, MapAccess, Visitor};
+use serde::Deserialize;
+
+const SECONDS_PER_MINUTE: u64 = 60;
+const SECONDS_PER_HOUR: u64 = 60 * SECONDS_PER_MINUTE;
+const SECONDS_PER_DAY: u64 = 24 * SECONDS_PER_HOUR;
+const SECONDS_PER_WEEK: u64 = 7 * SECONDS_PER_DAY;
+
+pub(crate) struct StringMapDeserializer<'de> {
+    iter: std::collections::hash_map::Iter<'de, String, String>,
+}
+
+impl<'de> StringMapDeserializer<'de> {
+    pub(crate) fn new(map: &'de HashMap<String, String>) -> Self {
+        StringMapDeserializer { iter: map.iter() }
+    }
+}
+
+impl<'de> Deserializer<'de> for StringMapDeserializer<'de> {
+    type Error = de::value::Error;
+
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
```

Comment:
If clippy is willing to leave it as a single line, you might consider the alternative:
```suggestion
    fn deserialize_any<V: Visitor<'de>>(self, visitor: V) -> Result<V::Value, Self::Error> {
```
(but also fine to keep it this way for consistency sake)

Reply:
(deleted)